### PR TITLE
Artificial data for non_uniform_axes

### PR DESCRIPTION
**Pull request by LMSC-NTappy**
### Description of the change
I implemented functions to generate two artificial datasets, a single spectrum and a 10x10 spectral map, reasonably similar to the luminescence of a semiconductor band-edge (it is actually just a gaussian). I implemented it following the style of the already existing artificial data series.

I think having such functions would help most testing endeavors in the development of non_uniform_axes (see #2740 and #2739)

### Progress of the PR
- [x] Change implemented (can be split into several points),
- [x] update docstring (if appropriate),
- [x] add tests,
- [x] ready for review.

### Minimal example of the bug fix or the new feature
```python
import hyperspy.api as hs
import  hyperspy.datasets.artificial_data as ad
s = ad.get_luminescence_map_nonuniform(uniform=False,add_baseline=False,add_noise=False)
s.plot()
```

### CODE REVIEW

File: hyperspy/datasets/artificial_data.py [hunk 1]
```diff
@@ -356,3 +363,231 @@ def get_atomic_resolution_tem_signal2d():
...
+    data = data.reshape((10,10,1024))
+    sig = Signal1D(data,axes=[spax_y,spax_x,nm_axis])
+    #sig.metadata.General.title = 'Artificial Luminescence map'
+
+    #if not uniform, transformation into non-linear axis
+    if not uniform:
+        hc = 1239.84198 #nm/eV
+
+        #eV axis creation. Note how the slice is inverted to have low energy first
+        evax = FunctionalDataAxis(expression="a/x",
```
Comment (ericpre) [inline hyperspy/datasets/artificial_data.py:575]: Would it be simpler to use `convert_to_functional_data_axis`?

File: hyperspy/datasets/artificial_data.py [hunk 2]
```diff
@@ -356,3 +363,231 @@ def get_atomic_resolution_tem_signal2d():
...
+    --------
+    get_low_loss_eels_signal,
+    get_core_loss_eels_signal,
+    get_low_loss_eels_line_scan_signal,
+    get_core_loss_eels_line_scan_signal,
+    get_core_loss_eels_model,
+    get_atomic_resolution_tem_signal2d,
+    get_luminescence_spectrum_nonuniform
+    """
+    from hyperspy.signals import Signal1D
```
Comment (ericpre) [inline hyperspy/datasets/artificial_data.py:511]: Unless strictly necessary, the import should be at the top of the file. Can you move them to the top?

File: hyperspy/datasets/artificial_data.py [hunk 3]
```diff
@@ -356,3 +363,231 @@ def get_atomic_resolution_tem_signal2d():
 
     s = Signal2D(image)
     return s
+
+
+def get_luminescence_spectrum_nonuniform(uniform=False, add_baseline=False, add_noise=True,random_state=None):
```
Comment (ericpre) [inline hyperspy/datasets/artificial_data.py:368]: To reduce duplication of code, an alternative would be to adapt and re-use `get_luminescence_map_nonuniform` use to covert this function. Maybe make rename `get_luminescence_map_nonuniform` to `get_luminescence_signal` and add a `navigation_dimension` argument to `get_luminescence_signal` to cover the behaviour of both functions?

**Review by ericpre — COMMENTED** [review]
Indeed, this would be useful! I have added a few suggestions for improvement.

Also, there are a couple of files, which should not be in this PR. The easiest may be to create a new branch starting from latest upstream and "cherry-pick" the relevant commits of this PR. Rebasing may not work well here.

File: hyperspy/datasets/artificial_data.py (hunk 3 above)
Comment (LMSC-NTappy) [reply hyperspy/datasets/artificial_data.py:368]: I agree. Will do

File: hyperspy/datasets/artificial_data.py (hunk 2 above)
Comment (LMSC-NTappy) [reply hyperspy/datasets/artificial_data.py:511]: OK I did this by imitation of the other artificial datasets function. Will move it to the top.

File: hyperspy/datasets/artificial_data.py (hunk 1 above)
Comment (LMSC-NTappy) [reply hyperspy/datasets/artificial_data.py:575]: It was. I did

**Comment by LMSC-NTappy** [comment]
> 
> 
> Indeed, this would be useful! I have added a few suggestions for improvement.
> 
> Also, there are a couple of files, which should not be in this PR. The easiest may be to create a new branch starting from latest upstream and "cherry-pick" the relevant commits of this PR. Rebasing may not work well here.

I have created another PR to fix this. See #2742

**Pull request closed without merging**